### ISSUE TITLE
anonymous_sender, Message-ID, and spam filtering

### ISSUE THREAD

**Issue by taggart** [sympa-community/sympa#1310]
Version
-------
All versions

Installation method
-------------------
All methods

Expected behavior
-----------------
The long existing anonymous_sender feature modifies the Message-ID of the emails before sending it out, so as to not reveal details about the original sender. A user [reported on the sympa-users list](https://listes.renater.fr/sympa/arc/sympa-users/2021-12/msg00011.html) that there are Spamassassin rules that compare headers like Message-ID and User-Agent to detect spammers forging headers. For example, emails with a legit Mozilla user agent always follow a particular pattern for Message-ID and if they aren't then it's likely spam.

Actual behavior
---------------
Lists using anonymous_sender are having their messages blocked by sites using default Spamassassin rulesets.

Additional information
----------------------
[More details about the reported case here](https://listes.renater.fr/sympa/arc/sympa-users/2021-12/msg00012.html). Maybe if sympa is modifying Message-ID, User-Agent should be stripped? Maybe User-Agent should _always_ be stripped, since sympa has effectively become the User-Agent as the message passes through?

On a system with spamassassin installed, looking for Message-ID in the rules is interesting:
`/var/lib/spamassassin$ rgrep Message-ID`

**Comment by ikedas** [comment]
Duplicate of #1306.

**Comment by ikedas** [comment]
@taggart, thanks for reporting.  I forgot to let you know I've submitted this issue.

**Comment by taggart** [comment]
Great! nearly identical bug report too! Thanks
